kernel/src/hil/sensors.rs

```diff
@@ -35,6 +35,21 @@ pub trait HumidityClient {
     fn callback(&self, value: usize);
 }
 
+/// A basic interface for a moisture sensor
+pub trait MoistureDriver<'a> {
+    fn set_client(&self, client: &'a dyn MoistureClient);
+    fn read_moisture(&self) -> Result<(), ErrorCode>;
+}
+
+/// Client for receiving moisture readings.
+pub trait MoistureClient {
+    /// Called when a moisture reading has completed.
+    ///
+    /// - `value`: the most recently read moisture in hundredths of
+    /// percent, or Err on failure.
```

Comment:
The HIL is simple enough, but what are the error types that this can return? I don't know whether there's anything special about the moisture sensor, so perhaps we can introduce (if it doesn't already exist) and refer to a standard list of error variants that all of these types of HILs can return (e.g., `ErrorCode::OFF`, which has pretty clear semantics across all sensors).

Reply:
I added documentation on the possible errors